apidoc/validate.py

```diff
@@ -1,623 +0,0 @@
-#!/usr/bin/env python
```

Comment:
I'm taking a look now. Looks like a few key differences are that the python script:
- enforced some properties were required on some structures (typically: name, summary)
- enforced a couple properties were valid markdown (not just strings)
- enforce properties whose name were all-caps had permission explicitly set to 'read-only'
- complained the 'notes' fields were '"no longer valid"

Related: should our build process be running the doc validation at some point? Sounds like maybe we should be doing this in an automated way to reject PRs with invalid doc changes...